packages/babel-plugin-transform-classes/src/vanilla.js

```diff
@@ -163,10 +163,24 @@ export default class ClassTransformer {
     //
     body.push(t.returnStatement(t.cloneNode(this.classRef)));
 
+    const parent = this.path.findParent(
+      path => path.isProgram() || path.isBlockStatement(),
+    );
+    const parentHasUseStrict = parent.node.directives.some(
+      directive => directive.value.value === "use strict",
+    );
+
+    const directives = [];
+
+    if (!parentHasUseStrict) {
+      directives.push(t.directive(t.directiveLiteral("use strict")));
+    }
```

Comment:
Nit: we are matching two times the `use strict` string, we could construct the node `t.directiveLiteral("use strict")` before. Just an idea.